edx-solutions/xblock-drag-and-drop-v2
# Display grade information

## Description

Student's grade was not displayed on DnDv2 problems. This patch adds code to display score information beneath the display name of the problem (similar to CAPA).

It also implements the missing `max_score` method which is required by the LMS to correctly calculate and display grades on the "Progress" tab.

![screen shot 2016-10-20 at 10 10 22](https://cloud.githubusercontent.com/assets/32585/19551847/8150b9ca-96ad-11e6-86f9-612651fc91d9.png)
## JIRA
- [SOL-2030](https://openedx.atlassian.net/browse/SOL-2030)
- [SOL-2094](https://openedx.atlassian.net/browse/SOL-2094)
## Sandbox URLs

Latest commit on the sandboxe: 3ad5a184dfe6ecb6644603cc75973e19738a97da
- [DnD unit in LMS](http://pr13797.sandbox.opencraft.hosting/courses/course-v1:DnDX+DnD101+2016/courseware/ce77c57034524ffb938c2b16e8316b7d/d49d04ffc18f4e738229aab063234adf/1?activate_block_id=block-v1%3ADnDX%2BDnD101%2B2016%2Btype%40vertical%2Bblock%40631785c086ab45278755c34fa7562a31)
- [Progress page in LMS](http://pr13797.sandbox.opencraft.hosting/courses/course-v1:DnDX+DnD101+2016/progress)
- [DnD unit in Studio](http://studio-pr13797.sandbox.opencraft.hosting/container/block-v1:DnDX+DnD101+2016+type@vertical+block@631785c086ab45278755c34fa7562a31)
## Testing instructions
1. Add a new DnDv2 block using default settings (standard mode).
2. Open the problem in LMS. The grey line below the title should say: "1 point possible (graded)" or "1 point possible (ungraded)", depending on whether you created the problem in a graded or ungraded section.
3. Toggle section between graded and ungraded (in Course Outline view in Studio, click the settings icon of the subsection containing the DnDv2 problem and select "Not Graded" or any of the other options if you want the problem to be graded).
4. Refresh the page in LMS. Make sure the text changes between "(graded)" <-> "(ungraded)".
5. Open the "Progress" page. Verify that the section corresponding to the DnD problem says `0/1`.
6. Go back to the DnD problem and start dragging items to the board. As you drop each item, check that the grade updates accordingly. Also check that the grade on the "Progress" page is always in sync with the grade displayed on the DnD problem.
7. Reset the problem. Verify that resetting the problem does not reset the grade (both on the problem itself and on the "Progress" page).
8. Change max score (weight) setting in Studio. Delete student state to clear the existing grade, then check that the grades are scaled correctly for max_score.
9. Repeat the above steps with a DnD problem in "assessment" mode. In assessment mode, the grade should only update when you explicitly submit the problem, not on every drop like it does in standard mode.
## Notes and Concerns
#96 makes decoy items contribute to the final score. This has the surprising effect that the first move you do on a fresh DnD problem that includes at least one decoy item is wrong, you immediately get awarded points for the decoy item.
## Reviewers
- [x] OpenCraft: @itsjeyd
- [ ] a11y: @edx/edx-accessibility
